tutorials/animation/animation_tree.rst

```diff
@@ -220,31 +230,137 @@ This is similar to 2D blend spaces, but in one dimension (so triangles are not n
+
+Here is the same example, but using Advance Expression rather than Advance Condition, which eliminates the need for two variables:
+
+.. image:: img/animtree_goodanimexpression.webp
+.. image:: img/animtree_goodanimexpression2.webp
+.. image:: img/animtree_goodanimexpression.gif
+
+In order to use Advance Expressions, the Advance Expression Base Node has to be set from the Inspector of the AnimationTree node. By default, it is set
+to the AnimationTree node itself, but it needs to point to whatever node contains the script with your animation variables.
+
+StateMachine travel
+^^^^^^^^^^^^^^^^^^^
+
+One of the nice features in Godot's ``StateMachine`` implementation is the ability to travel. You can instruct the graph to go from the
+current state to another one, while visiting all the intermediate ones. This is done via the A\* algorithm.
+If there is no path of transitions starting at the current state and finishing at the destination state, the graph teleports to the destination state.
+
+To use the travel ability, you should first retrieve the :ref:`AnimationNodeStateMachinePlayback <class_AnimationNodeStateMachinePlayback>`
+object from the ``AnimationTree`` node (it is exported as a property), and then call one of its many functions:
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    var state_machine = animation_tree["parameters/playback"]
+    state_machine.travel("SomeState")
+
+ .. code-tab:: csharp
+
+    AnimationNodeStateMachinePlayback stateMachine = (AnimationNodeStateMachinePlayback)animationTree.Get("parameters/playback");
+    stateMachine.Travel("SomeState");
+
+The StateMachine must be running before you can travel. Make sure to either call ``start()`` or connect a node to **Start**.
+
+BlendSpace2D and BlendSpace1D
+-----------------------------
+
+``BlendSpace2D`` is a node to do advanced blending in two dimensions. Points representing animations are added to a 2D space and then a position between them
+is controlled to determine the blending:
+
+.. image:: img/animtree_blendspace2d.gif
+
+You may place these points anywhere on the graph by right clicking or using the **add point** button, whose icon is a pen and point. 
+Wherever you place the points, the triangle between them will be generated automatically using Delaunay.
+You may also control and label the ranges in X and Y. 
+
+.. image:: img/animtree_blendspacepoints.gif
+
+Finally, you may also change the blend mode. By default, blending happens by interpolating points inside the closest triangle. When dealing with 2D 
+animations (frame by frame), you may want to switch to *Discrete* mode. Alternatively, if you want to keep the current play position when switching 
+between discrete animations, there is a *Carry* mode. This mode can be changed in the *Blend* menu:
+
+.. image:: img/animtree_blendmode.webp
+
+BlendSpace1D works just like BlendSpace2D, but in one dimension (a line). Triangles are not used.
+
+.. image:: img/animtree_blendspace1d.webp
```

Comment:
There's existing sections about blendspace 1D and 2D earlier in the page, not sure what happened here, or if something wasn't removed that was supposed to be.